Lombiq.HelpfulLibraries.OrchardCore/Security/SkipContentSecurityPolicyProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Lombiq.HelpfulLibraries.AspNetCore.Security;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Options;
+using OrchardCore.Admin;
+using System;
+using System.Net.Mime;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// A provider for some common scenarios where the header should not be applied at all.
+/// </summary>
+public class SkipContentSecurityPolicyProvider : IContentSecurityPolicyProvider
+{
+    private readonly string _adminPathPrefix;
+
+    public SkipContentSecurityPolicyProvider(IOptions<AdminOptions> adminOptions) =>
+        _adminPathPrefix = '/' + adminOptions.Value.AdminUrlPrefix;
+
+    public ValueTask<bool> ShouldSuppressHeaderAsync(HttpContext context) =>
+        new(ShouldSuppressHeaderInner(context));
+
+    private bool ShouldSuppressHeaderInner(HttpContext context) =>
+        // No need to do content security policy on non-HTML responses.
+        context.Response.ContentType?.ContainsOrdinalIgnoreCase(MediaTypeNames.Text.Html) != true ||
+        // The Admin dashboard is only accessible to trusted users so content security is not a concern.
+        new Uri(context.Request.GetDisplayUrl()).AbsolutePath.StartsWithOrdinalIgnoreCase(_adminPathPrefix);
```

Comment:
It can be a concern since you can do an XSS attack from the frontend, that'll be only active on the admin (like loading a script in a WYSIWYG editor). The admin should be guarded too.

Reply:
Well okay, but we have to permit eval due to the media field editor using Vue anyway so I'm a bit dubious how helpful that is.

Reply:
That can be allowed only under specific URLs.

Reply:
What do you mean? The URL would be the same `/Admin/Contents/ContentItems/{GUID}/Edit` where a WYSIWYG editor might appear.

Reply:
Yeah, that one. Doesn't the Media Library need that too, BTW?

Reply:
Yep. Did that in TDEAL

Reply:
Why not here too?

Reply:
Because it's kludge.

Reply:
Can you elaborate? Won't we have exactly the same issue in all of our projects using Media and the HL security default?

Reply:
I'm not expecting that, considering that we have yet to find a whole second person who can manifest the problem in question. I works flawlessly for you, me and PG, nobody else have yet complained about it in OCC, it was never a problem in UI tests either. I'd rather have more data points.